src/makeCalculator.js

```diff
@@ -4,7 +4,31 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  return {
+    result: 0,
+    reset() {
+      this.result = 0;
+
+      return this;
+    },
+    operate(callback, number) {
+      callback(number, this);
+
+      return this;
```

Comment:
The `operate` method should ensure that the provided callback is a function before attempting to call it. This can be done by checking `typeof callback === 'function'`.